### PR TITLE
Clarify delegated role verification on primary ECUs. Closes #36

### LINKED ISSUE #36: Full Verification algorithm does not verify delegated roles (5.4.4.2:11.2)
While the algorithm in 5.4.4.2 seems to correctly capture verification of the top-level metadata roles (root, timestamp, snapshot, targets), **it does not seem to include verification for delegated targets roles**, which must be verified based on a traversal of the graph of delegations in step 11.2.

There's a lot of discussion about this [in TUF issue 660](https://github.com/theupdateframework/tuf/issues/660), if you want a deeper dive.  I'll try for a short version:

The naive and incorrect way to verify delegated roles is to verify the role when it is first needed, based on the delegation that first brought you to it, and to treat the role as verified from then on.  This treats it like a top-level role (timestamp, snapshot, targets, root).  The trouble is that top-level roles can always be verified based on what the root role says, but delegated roles don't have a single definitive source for the information necessary to verify them.  Two different delegated roles A and B can both delegate to role C, and those two separate delegations can indicate different paths (and/or hardware IDs) and indicate different thresholds and sets of keys (Perhaps one is a subset of the other, perhaps not.).  There is no meaning to the notion of verifying delegated role C except in the context of a particular path through the delegation graph, and that verification can look different if you come to it from role A vs role B first.  If you want target info for image ```1.tgz```, you walk the graph of delegations and can verify each role you encounter based on the delegation that brought you to it.  If you're not walking the graph, there are no rules for verifying delegated roles (unlike the top-level roles, which are all specified directly by root).

So the naive and correct and inefficient way to do this is to verify a delegated role every time you encounter it while traversing the delegation graph.  You could improve this by caching verification status indexed by delegation paths, or in any number of ways, but you cannot assume that every delegated role file can be verified once regardless of the structure of the delegation graph.

### PR DESCRIPTION
**Pull request by tkfu**
This clearly overlaps with the discussions in #71 and #33, but I'm trying to make the narrowest change that addresses the specific issue in #36. The details of where hardware IDs and ECU IDs are checked can be changed in a future PR once we have made a decision about those issues.

### CODE REVIEW

File: uptane-standard.md [hunk 1]
```diff
@@ -860,7 +860,8 @@ In order to perform full verification, an ECU SHALL perform the following steps:
 8. Download and check the Timestamp metadata file from the Image repository, following the procedure in {{check_timestamp}}.
 9. Download and check the Snapshot metadata file from the Image repository, following the procedure in {{check_snapshot}}.
 10. Download and check the top-level Targets metadata file from the Image repository, following the procedure in {{check_targets}}.
-11. For each image listed in the Targets metadata file from the Director repository downloaded in step 6, complete the following procedure:
+11. Determine which image(s) to verify metadata for. If this is a primary ECU, verify metadata for all images listed in the Targets metadata file from the Director repository downloaded in step 6. If this is a secondary ECU, verify only the metadata for the image that includes the ECU identifier for this ECU.
```
Comment (trishankatdatadog) [inline uptane-standard.md]: For the secondary ECU, don't you just mean the Targets metadata file from the director repository?

**Comment by trishankatdatadog** [comment]
I don't understand what this PR is trying to clarify exactly. Could you explain, please?

File: uptane-standard.md (hunk 1 above)
Comment (tkfu) [reply uptane-standard.md]: I mean Targets metadata from both image and director--re-reading this, I think I can make it a bit clearer.

**Comment by tkfu** [comment]
> I don't understand what this PR is trying to clarify exactly. Could you explain, please?

It's specifying exactly how the primary ECU verifies targets metadata for secondaries. As @awwad pointed out in #36, it's potentially possible to reach a delegated role from more than one path, and the correct signatures to validate for a delegated role may differ based on how it was reached.

By specifying that the primary MUST do delegation resolution for each image listed in the Director targets metadata, we make sure that each role is being reached in the correct way, and thus validated with the correct keys. I'll work on making this a bit clearer.

**Comment by trishankatdatadog** [comment]
Got it, thanks.

The part about multiple ways to reach the "same" role with different keys has always been there, and it's nothing new. As long one does preorder DFS correctly, this shouldn't be an issue.

It's a TUF reference implementation issue that doesn't handle getting these keys correctly yet, IIRC. Not anything that should terribly concern us here, although I guess we should clarify that the same role can have different keys depending on who's delegating to it and how you got there.

**Comment by jhdalek55** [comment]
At our 3/27 meeting it was noted that PR#81 will resolve this issue. That PR will be merged as soon as @tkfu  addresses the comments of @trishankatdatadog.

File: uptane-standard.md [hunk 2]
```diff
@@ -860,16 +860,16 @@ In order to perform full verification, an ECU SHALL perform the following steps:
 8. Download and check the Timestamp metadata file from the Image repository, following the procedure in {{check_timestamp}}.
 9. Download and check the Snapshot metadata file from the Image repository, following the procedure in {{check_snapshot}}.
 10. Download and check the top-level Targets metadata file from the Image repository, following the procedure in {{check_targets}}.
-11. For each image listed in the Targets metadata file from the Director repository downloaded in step 6, complete the following procedure:
-    1. Locate and download a Targets metadata file that contains an image with exactly the same file name, following the procedure in {{resolve_delegations}}.
+11. Verify that Targets metadata from the Director and Image repositories match. A primary ECU MUST perform this check on metadata for all images listed in the Targets metadata file from the Director repository downloaded in step 6. A secondary ECU MAY elect to perform this check only on the metadata for the image it will install. (That is, the metadata from the Director that contains the ECU identifier of the current ECU.) To check that the metadata for an image matches, complete the following procedure:
```
Comment (patrickvacek) [inline uptane-standard.md]: Looks fine to me except that the parenthetical text is a sentence fragment, which threw me off for a minute.

Comment (trishankatdatadog) [inline uptane-standard.md]: If we could just add: "(That is, the **target** metadata from the Director that contains..."

**Pull request merged**